pkg/server/container_execsync.go

```diff
@@ -140,40 +131,33 @@ func (c *criContainerdService) execInContainer(ctx context.Context, id string, o
 		}
 	})
 
-	resCh := make(chan execResult, 1)
-	go func() {
-		// Wait will return if context is cancelled.
-		exitCode, err := process.Wait(ctx)
-		resCh <- execResult{
-			exitCode: exitCode,
-			err:      err,
-		}
-		glog.V(2).Infof("Exec process %q exits with exit code %d and error %v", execID, exitCode, err)
-	}()
+	exitCh, err := process.Wait(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to wait for process %v", err)
```

Comment:
And let's make the error different from the one below, so that once there is an error, we know which lines feels when debugging.

How about:
* Here: `failed to start waiting for exec %q: %v`
* Below: `failed when waiting for exec %q: %v`

Something like this, your English is better than mine. :)